connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -535,6 +535,159 @@ class Dataset[T] private[sql] (
     }
   }
 
+  /**
+   * Gets the Dataset as string in a tabular form. Strings more than 20 characters will be
+   * truncated, and all cells will be aligned right. For example:
+   * {{{
+   *   year  month AVG('Adj Close) MAX('Adj Close)
+   *   1980  12    0.503218        0.595103
+   *   1981  01    0.523289        0.570307
+   *   1982  02    0.436504        0.475256
+   *   1983  03    0.410516        0.442194
+   *   1984  04    0.450090        0.483521
+   * }}}
+   *
+   * @param numRows
+   *   Number of rows to show
+   * @group action
+   * @since 3.4.0
```

Comment:
nit: Why is this `3.4.0` while the latter changes are marked as `3.5.0`?

Reply:
This was added on its own. not sure why?

> Hi @VindhyaG, this might be useful - may be we can benefit from the usecase you have for this. Is it just for logging? Not sure what others think, it might be good to limit the API surface.



> I missed changing version number . will change once i can figure out what is the current release cycle is on? Can you please point me where i can get that info .thanks!



> Hi @VindhyaG, this might be useful - may be we can benefit from the usecase you have for this. Is it just for logging? Not sure what others think, it might be good to limit the API surface.



> Hi @VindhyaG, this might be useful - may be we can benefit from the usecase you have for this. Is it just for logging? Not sure what others think, it might be good to limit the API surface.

@ScrapCodes Thanks for reviewing.  bug  lists the three reasons which i have listed in PR description above. Do you have any specific suggestions to limit API surface?

Reply:
@jaceklaskowski updated to 3.5.0 based on recently(hours ago) closed bug fix version of SQL and Java API component